src/components/MoneyRequestConfirmationList.js

```diff
@@ -750,14 +750,8 @@ function MoneyRequestConfirmationList(props) {
                             }}
                             disabled={didConfirm}
                             interactive={!props.isReadOnly}
-                            brickRoadIndicator={
-                                props.isPolicyExpenseChat && shouldDisplayFieldError && TransactionUtils.isMerchantMissing(transaction) ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : ''
-                            }
-                            error={
-                                shouldDisplayMerchantError || (props.isPolicyExpenseChat && shouldDisplayFieldError && TransactionUtils.isMerchantMissing(transaction))
-                                    ? translate('common.error.enterMerchant')
-                                    : ''
-                            }
+                            brickRoadIndicator={shouldDisplayMerchantError ? CONST.BRICK_ROAD_INDICATOR_STATUS.ERROR : ''}
+                            error={shouldDisplayMerchantError ? translate('common.error.enterMerchant') : ''}
```

Comment:
The condition is simplified here. We don't need to check `TransactionUtils.isMerchantMissing` anymore because `shouldDisplayMerchantError` already contains `isMerchantEmpty`.

`props.iouMerchant` contains either `modifiedMerchant` or `merchant`.

Also, the condition here is last updated in this [PR](https://github.com/Expensify/App/pull/35112), I was trying to make sure the test step there still passed, but we have this [issue](https://github.com/Expensify/App/issues/35667#issuecomment-1938360799)